lib/modules/manager/devcontainer/extract.spec.ts

```diff
@@ -0,0 +1,363 @@
+        }
+      }`);
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(
+        fixture.content,
+        fixture.path,
+        extractConfig,
+      );
+
+      // Assert
+      expect(result).not.toBeNull();
+      expect(result?.deps).not.toBeNull();
+      expect(result?.deps.length).toBe(2);
+      expect(result?.deps).toMatchInlineSnapshot(`
+      [
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "1.2.3",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/features/first",
+          "replaceString": "devcontainer.registry.renovate.com/test/features/first:1.2.3",
+        },
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "4.5.6",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/features/second",
+          "replaceString": "devcontainer.registry.renovate.com/test/features/second:4.5.6",
+        },
+      ]
+      `);
+    });
+
+    it('returns image and feature image deps when both image and features properties are defined in dev container JSON file', () => {
+      // Arrange
+      const fixture = getFixture(`
+      {
+        "image": "devcontainer.registry.renovate.com/test/image:1.2.3",
+        "features": {
+          "devcontainer.registry.renovate.com/test/feature:4.5.6": {}
+        }
+      }`);
+      const extractConfig = {};
+
+      // Act
+      const result = extractPackageFile(
+        fixture.content,
+        fixture.path,
+        extractConfig,
+      );
+
+      // Assert
+      expect(result).not.toBeNull();
+      expect(result?.deps).not.toBeNull();
+      expect(result?.deps.length).toBe(2);
+      expect(result?.deps).toMatchInlineSnapshot(`
+      [
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "1.2.3",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/image",
+          "replaceString": "devcontainer.registry.renovate.com/test/image:1.2.3",
+        },
+        {
+          "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
+          "currentDigest": undefined,
+          "currentValue": "4.5.6",
+          "datasource": "docker",
+          "depName": "devcontainer.registry.renovate.com/test/feature",
+          "replaceString": "devcontainer.registry.renovate.com/test/feature:4.5.6",
+        },
+      ]
+      `);
```

Comment:
Same

Reply:
I've removed the initial assertions but changing `toMatchInlineSnapshot` to `toEqual` gets automatically changed to `toBe` and then fails because of the JSON formatting, so I've left it using `toMatchInlineSnapshot`.